python/lsst/afw/cameraGeom/cameraFactory.py

```diff
@@ -48,6 +48,28 @@ def makeDetector(detectorConfig, ampInfoCatalog, focalPlaneToPupil):
         transforms,
     )
 
+def copyDetector(detector, ampInfoCatalog=None):
```

Comment:
What is the point of the function is if `ampInfoCatalog` is omitted? This feels like API clutter. If the only use case is making a new detector for an untrimmed raw image then I suggest writing a function that does that (though most of the trimming work probably belongs in or near `AmpInfoRecord`, as I said in my other review). Then this code could moved there (either as a separate private function or incorporated into the body).